lib/sidekiq/worker.rb

```diff
@@ -191,6 +191,12 @@ def perform_async(*args)
         @klass.client_push(@opts.merge("args" => args, "class" => @klass))
       end
 
+      def perform_bulk(args, batch_size: 1_000)
+        args.each_slice(batch_size).flat_map do |slice|
+          Sidekiq::Client.push_bulk(@opts.merge("class" => @klass, "args" => slice))
```

Comment:
@mperham this is a bit different than the example on https://github.com/mperham/sidekiq/wiki/Batches#huge-batches in that the jobs get sequentially added by push_bulk rather than concurrently added via perform_async. On super large sets of jobs, like inserting 300,000 at once in 500 job push_bulks with about 400 worker threads, Redis times out sometimes. Maybe the recommendation for huge batches should be changed?

I suspect that it's just as good performance-wise to sequentially post the jobs by bulk by a single thread.

Reply:
I don't understand.

Reply:
![image](https://user-images.githubusercontent.com/1118459/155036707-1c2e34be-9fbf-4092-a51f-aff920573c61.png)

and got many errors: 

![image](https://user-images.githubusercontent.com/1118459/155036851-48e2b585-e2e2-4951-bf73-5b028a7355b8.png)

Reply:
Yeah, it’s not a great idea to do lots of concurrent bulk pushes. I think the best thing you can do is lower the bulk size from 1000 to 100 so each bulk op is faster but YMMV.

Reply:
Maybe better to do the bulk inserts sequentially rather than concurrently? Especially if running hundreds of threads.